feat: Add xms_edov claim for nOAuth vulnerability (#220)
clerkinc/clerk-docs: docs/authentication/social-connections/microsoft.mdx

![](/docs/images/authentication-providers/microsoft/08f6445903a233c89ad96be25de3293def891cb6-1172x854.png)

> After confirming your choice the new secret will be listed. This is the only time you will able to copy & paste it, so make sure you do so before navigating away. Also, take note of the lifetime of said secret because you will need to generate a new one before it expires.

Reviewer: ```suggestion
After confirming your choice, the new secret will be listed. This is the only time you will able to copy & paste it, so make sure you do so before navigating away. Also, take note of the lifetime of said secret because you will need to generate a new one before it expires.

```

Congratulations! Social connection with Microsoft Azure is now configured for your instance.

Stay tuned for updates as Clerk integrates support for more Microsoft social connection options.

### Stay secure against the nOAuth vulnerability

A vulnerability called [nOAuth](https://www.descope.com/blog/post/noauth) has been recently discovered in Microsoft's Azure AD OAuth applications. This vulnerability can lead to full account takeovers, as attackers can spoof the provided email and potentially gain full control of a victim's account.

To protect users, Clerk has implemented stringent measures against the Microsoft OAuth provider, enforcing stricter checks on verified email addresses. However, this has impacted the functionality of users using Azure AD.

Microsoft has introduced an optional claim, `xms_edov`, that can be added to your Azure application. This claim provides additional context to determine whether the returned email is verified.

To enable this optional claim, you'll need to follow these steps:

> Navigate to your Azure application's **"Token configuration"** page, and click on **"Add optional claim"**.

Reviewer: ```suggestion
Navigate to your Azure application's **Token configuration** page, and click on **Add optional claim**.

```

![Add optional claims](/docs/images/authentication-providers/microsoft/noauth-optional-claims-init.png)

> Select the `email` and `xms_pdl` claims for both **ID** and **Access**, and click on **"Add"**.

Reviewer: ```suggestion
Select the `email` and `xms_pdl` claims for both **ID** and **Access**, and click on **Add**.

```

> __Note: At the time of writing, the `xms_edov` claim is still in preview, and may not be available for all applications. So we'll choose another one from the list and we'll rename it later on in the manifest.__

Reviewer: ```suggestion
<Callout type="info">
At the time of writing, the `xms_edov` claim is still in preview, and may not be available for all applications. So we'll choose another one from the list and we'll rename it later on in the manifest.
</Callout>
```

![Select claims](/docs/images/authentication-providers/microsoft/noauth-optional-claims-select.png)

Accept the permission changes. This will add `email` in your API permissions.

![Accept permissions](/docs/images/authentication-providers/microsoft/noauth-optional-claims-permissions.png)

> Navigate to the **"Manifest"** page, and update the `idToken` and `accessToken` values names from `xms_pdl` to `xms_edov`.

Reviewer: ```suggestion
Navigate to the **Manifest** page, and update the `idToken` and `accessToken` values names from `xms_pdl` to `xms_edov`.

```

![Update manifest](/docs/images/authentication-providers/microsoft/noauth-optional-claims-manifest.png)

> Navigate back to the **"Token configuration"** page, and check the `xms_edov` claim for both **ID** and **Access**.

Reviewer: ```suggestion
Navigate back to the **Token configuration** page, and check the `xms_edov` claim for both **ID** and **Access**.

```

![Check claims](/docs/images/authentication-providers/microsoft/noauth-optional-claims-check.png)

> Now you're all set! Now Microsoft will send the `xms_edov` claim in the token, and Clerk will use it to determine whether the email is verified or not, even if it is used with Azure AD.

Reviewer: ```suggestion
Now you're all set! Microsoft will send the `xms_edov` claim in the token, and Clerk will use it to determine whether the email is verified or not, even if it is used with Azure AD.
```

### Stay secure against the nOAuth vulnerability

A vulnerability called [nOAuth](https://www.descope.com/blog/post/noauth) has been recently discovered in Microsoft's Azure AD OAuth applications. This vulnerability can lead to full account takeovers, as attackers can spoof the provided email and potentially gain full control of a victim's account.

To protect users, Clerk has implemented stringent measures against the Microsoft OAuth provider, enforcing stricter checks on verified email addresses. However, this has impacted the functionality of users using Azure AD.

Microsoft has introduced an optional claim, `xms_edov`, that can be added to your Azure application. This claim provides additional context to determine whether the returned email is verified.

To enable this optional claim, you'll need to follow these steps:

Navigate to your Azure application's **Token configuration** page, and click on **Add optional claim**.

![Add optional claims](/docs/images/authentication-providers/microsoft/noauth-optional-claims-init.png)

Select the `email` and `xms_pdl` claims for both **ID** and **Access**, and click on **Add**.

<Callout type="info">

>     At the time of writing, the `xms_edov` claim is still in preview, and may not be available for all applications. So
>     we'll choose another one from the list and we'll rename it later on in the manifest.

Reviewer: ```suggestion
    At the time of writing, the `xms_edov` claim is still in preview, and may not be available for all applications. So we'll choose another one from the list and we'll rename it later on in the manifest.

```

Author: Hmm, looks like IDE splits the lines on format because it parses this as a component. Anyway, it's fixed – thanks!
